tests/test_ai_sidecar_external.py

```
@@ -0,0 +1,247 @@
+
+
+def test_upgrade_accepted_with_token() -> None:
+    async def _run() -> dict:
+        async with _running_sidecar(token="s3cret") as port:
+            async with ws_connect(
+                f"ws://127.0.0.1:{port}/",
+                additional_headers={
+                    ep.AUTH_HEADER: "s3cret",
+                    ep.CLIENT_HEADER: "test-client",
+                },
+            ) as ws:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)